src/Repository/CarrinhoRepository.php

```diff
@@ -16,12 +17,27 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, Carrinho::class);
     }
 
-    public function salvar(Carrinho $carrinho): Carrinho
+    public function salvar(Carrinho $carrinho): int
     {
         $this->getEntityManager()->persist($carrinho);
         $this->getEntityManager()->flush();
         
-        return $carrinho;
+        return $carrinho->getId();
```

Comment:
a ideia de já retornar o Carrinho é você usar o id se precisar, então retorna o Carrinho aqui e não o id

```suggestion
        return $carrinho;
```